test/unit/hashes.test.js

```diff
@@ -17,3 +21,71 @@ describe('deobfuscation', function() {
     })
   })
 })
+
+describe('Buffer.from fallback logic', function() {
+  var fromCalled = false
+
+  function getBufferStub(shouldFail) {
+    fromCalled = false
+
+    return xtend(Buffer, {
```

Comment:
We make a clone of Buffer and stub the `from` function with a test implementation. I considered using `sinon` but wanted to add as few devDependencies as possible.